lib/rules/static-jsx.js

```diff
@@ -17,19 +21,12 @@ module.exports = {
       recommended: true,
     },
     fixable: 'code',
+    messages,
     schema: [],
   },
 
   create: function (context) {
-    const curlyBracePresenceInstance = curlyBracePresence.create({
-      ...context,
-      options: [
-        {
-          props: 'never',
-          children: 'ignore',
-        },
-      ],
-    })
+    const curlyBracePresenceInstance = curlyBracePresence.create(context)
```

Comment:
Not sure why but if we pass `{...context}` instead of `context` - we are loosing `context` methods like `context.getSourceCode()` and I was getting errors like

```
  ● static-jsx › invalid › (<FormattedMessage id={'greeting'} defaultMessage={`hello`} />)

    TypeError: context.getSourceCode is not a function
    Occurred while linting <input>:1

      52 |             attrNode.value.type === 'JSXExpressionContainer'
      53 |           ) {
    > 54 |             curlyBracePresenceInstance.JSXExpressionContainer(attrNode.value)
         |                                        ^
      55 |           }
      56 |         })
      57 |
```
Could not find docs what is the proper way to clone context and pass it to another rule.

Reply:
Another option that works but looks strange:

```
const curlyBracePresenceInstance = curlyBracePresence.create({
  ...context,
  // manually define getSourceCode() ... 
  getSourceCode: () => context.getSourceCode(),
  options: [
    {
      props: 'never',
      children: 'ignore',
    },
  ],
})
```